engine/src/main/java/org/terasology/monitoring/impl/PerformanceMonitorImpl.java

```diff
@@ -40,8 +41,10 @@
 public class PerformanceMonitorImpl implements PerformanceMonitorInternal {
     private static final int RETAINED_CYCLES = 60;
     private static final double DECAY_RATE = 0.98;
+    private static final Activity OFF_THREAD_ACTIVITY = new NullActivity();
```

Comment:
I wish I had commented on this before the PR was merged. I don't quite understand why this is done, why the necessity to have  a NullActivity for non-main-thread activities. But I wasn't going to ask about it, I was going to suggest to put the explanation in the class javadoc for NullActivity! Any chance you can write a couple of lines about it Immortius? If you want you can write them here and then I'll take care of submitting a PR for them.